added animated share buttons

danielfmiranda opened the pull request:
Closes #7985
Related PRs/issues #

**Link to sample test page**: https://foundation-s-animated-s-slhda2.herokuapp.com/en/youtube/regretsreporter/

**Steps to test**: 

1. Visit test page above
2. Click the share this button
3. buttons should animate as expected
4. Clicking on buttons should preform their expected functions with prepopulated copy, and the copy URL button should do so.
5. If they all work, testing is complete!

beccaklam commented:
These look _so_ good @danielfmiranda! 💯 The animation is perfect! Just one question, the share button icon remains selected even after I have deselected it. Only when I click somewhere else on the page does it deselect. Is there a way to make it so that once the mouse is no longer on it/hovering it deselects? Small detail though!

danielfmiranda commented:
Hi @beccaklam, thanks for the feedback! 
I have updated it to now only highlight when the user hover overs the button. 
I have also added the functionality for the share buttons, they should now behave as expected :D 

I do have one question however, I was wondering if its possible to get an icon for the "success" stage of the link copy button? 

Right now, when you copy the link it changes to a check mark to show it went through successfully, however since the icon isn't from the same set it looks too big for the button. 
Thank you!

TheoChevalier commented:
Hey folks, heads up that I’ve added changes to be merged in this branch to support localization here: https://github.com/mozilla/foundation.mozilla.org/pull/8012

They need to be merged before this PR is reviewed + merged

beccaklam reviewed (changes requested):
Hi Daniel, this is looking great and working well! ✨ Thank you again for making this happen!
The hover state for the share icon button is working perfectly now. 

**One change request:**
For the share link button, would we be able to style the "checked/copied" state to be the same as Blog's copied link icon? I think the check just needs to be smaller and centered. The circle around the check mark should also just stay the same. Right now I think it turns blue when you hover over it:
<img width="55" alt="Screen Shot 2021-12-15 at 6 17 44 AM" src="https://user-images.githubusercontent.com/20977376/146203037-8ab2c9ba-e117-473b-b044-e0eef9238d78.png">
Below is what Blog's copied link icon looks like (on the RegretsReporter landing page it would just be the inverse of this). You can see a live version [here](https://foundation.mozilla.org/en/blog/breaking-bias-2-when-twitters-negligence-harms-democracy/).
<img width="56" alt="Screen Shot 2021-12-15 at 6 13 43 AM" src="https://user-images.githubusercontent.com/20977376/146202828-f1f713d7-400a-4f57-818f-924ae63b01c0.png">
Thank you so much!

beccaklam commented:
Here's the checkmark icon you asked for @danielfmiranda ! Sorry for not reading your previous comment before submitting my change request 😅 
https://drive.google.com/open?id=1KvqwfOEDmhY5KdBKycHOOuFtT1ZAHJZV&authuser=rebeccal%40mozillafoundation.org&usp=drive_fs

beccaklam commented:
@danielfmiranda [Here is the check mark](https://drive.google.com/open?id=1L9_k-aiLgEOV5olp8vk-WRd963iE_UNf&authuser=rebeccal%40mozillafoundation.org&usp=drive_fs) icon you requested @danielfmiranda! Sorry for not reading your comment earlier before posting my change request!

fessehaye commented on source/js/foundation/pages/youtube-regrets/regrets-reporter/share-buttons.js:
```diff
@@ -0,0 +1,113 @@
+  // Display share buttons and hide "Share This" text
+  displayShareButtons() {
+    tl.to("#share-display-text", {
+      display: "none",
```
You can't 'animate' display. You want to animate the text and then add the the display none afterwards. Remove `display: "none",` from `tl.to` and then add a line afterwards using `tl.set` instead. If you do not want the element to be animated then just remove the other properties besides display none and use `tl.set` instead of `tl.to`

danielfmiranda replied:
Updated!

fessehaye commented on source/sass/views/regrets-reporter-landing-page.scss:
```diff
@@ -47,4 +47,78 @@
     height: 100%;
   }
+
+  .share-button-container {
```
Hmm I feel like you could have probably do a majority of this with tailwind classes including the background images as well instead of writing a sass file. I do not think it is required for the ticket but I think it would be a great refactor task in the future. Can totally walk through this with you as well!

danielfmiranda replied:
Would love to Simon! Going to take a note to make a new ticket to walk through this!

fessehaye commented on source/js/foundation/pages/youtube-regrets/regrets-reporter/share-buttons.js:
```diff
@@ -0,0 +1,113 @@
+        y: 10,
+        display: "none",
+        opacity: 0,
+        stagger: -0.1,
```
Should this stagger value be negative?

danielfmiranda replied:
Hi Simon, yes it is! To reverse the order of the buttons dissapearing

fessehaye commented on source/js/foundation/pages/youtube-regrets/regrets-reporter/share-buttons.js:
```diff
@@ -0,0 +1,113 @@
+        duration: 0.4,
+      }
+    );
+    tl.to("#share-display-text", {
```
If you have multiple `gsap.to` calls in a row. It would be best to make a timeline to keep the animations in sync!

danielfmiranda replied:
Hi Simon! I believe I am doing this at the top of the file, creating a timeline called tl using gsap.timeline(). Will double check with you in our call soon!

fessehaye commented on source/js/foundation/pages/youtube-regrets/regrets-reporter/share-buttons.js:
```diff
@@ -0,0 +1,113 @@
+      {
+        y: 10,
+        display: "none",
+        opacity: 0,
```
replace `opacity` with `autoAlpha` in all gsap instances

danielfmiranda replied:
Done!

beccaklam reviewed (approved):
This is good to go! Thanks so much for your work on this @danielfmiranda!
